dts/riscv/openhwgroup/cva6.dtsi

```diff
@@ -0,0 +1,191 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2024 CISPA Helmholtz Center for Information Security
+ */
+
+/*
+ * CVA6 SoC without CPU, which comes in 32 and 64 bit variants
+ */
+
+/ {
+	#address-cells = <0x01>;
+	#size-cells = <0x01>;
+
+	clocks {
+
+		clk_cpu {
+			#clock-cells = <0x00>;
+			clock-frequency = <50000000>;
+			compatible = "fixed-clock";
+		};
+
+		clk_bus: clk_bus_0 {
+			#clock-cells = <0x00>;
+			clock-frequency = <50000000>;
+			compatible = "fixed-clock";
+		};
+	};
+
+	soc {
+		#address-cells = <0x01>;
+		#size-cells = <0x01>;
+		ranges;
+
+		/* Xilinx MIG memory controller */
+		memory0: memory@80000000 {
+			device_type = "memory";
+			reg = <0x80000000 0x40000000>;
+			status = "okay";
+		};
+
+		/* RISC-V Platform-level interrupt controller */
+		plic: interrupt-controller@c000000 {
+			compatible = "sifive,plic-1.0.0";
+			#address-cells = <0>;
+			#interrupt-cells = <2>;
+			interrupt-controller;
+			interrupts-extended = <&hlic 11 &hlic 9>;
+			reg = <0x0c000000 0x4000000>;
+			riscv,max-priority = <7>;
+			riscv,ndev = <30>;
+			status = "okay";
+		};
+
+		/* USB UART */
+		uart0:serial@10000000 {
+			clock-frequency = <50000000>;
+			clocks = <&clk_bus>;
+			compatible = "ns16550";
+			current-speed = <115200>;
+			device_type = "serial";
+			reg = <0x10000000 0x10000>;
+			reg-shift = <0x02>;
+			parity = "none";
+			stop-bits = "1";
+			data-bits = <8>;
+
+			interrupt-parent = <&plic>;
+			interrupts = <11 0x04>;
+
+			status="okay";
+		};
+
+		/* SPI controller, connected to SD card */
+		spi0: axi_quad_spi@20000000 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			compatible = "xlnx,xps-spi-2.00.a";
+			reg = <0x20000000 0x10000>;
+			xlnx,num-ss-bits = <0x01>;
+			xlnx,num-transfer-bits = <0x8>;
+			interrupts=<10 4>;
+			interrupt-parent=<&plic>;
+			clocks=<&clk_bus>;
+
+			status="okay";
+		};
+
+		/*
+		 * Core-local interrupt controller
+		 * Clic in zephyr terminology, clint in CVA6
+		 */
+		clint: clint_wrapper_verilog@2000000 {
+			compatible = "sifive,clint0";
+			reg = <0x2000000 0x40000>;
+			interrupts-extended = <&hlic 3 &hlic 7>;
+
+			status="okay";
+		};
+
+		/*
+		 * Xilinx AXI Ethernet subsystem and DMA.
+		 * Only on CISPA fork of the project.
+		 */
+		dma0: dma@41e00000 {
+			#dma-cells = <0x01>;
+			clock-frequency = <50000000>;
+			clock-names = "s_axi_lite_aclk";
+			clocks = <&clk_bus>;
+			compatible = "xlnx,eth-dma";
+			reg = <0x41e00000 0x10000>;
+			xlnx,addrwidth = <0x40>;
+			xlnx,include-dre;
+			xlnx,num-queues = <0x1>;
+
+			interrupt-parent = <&plic>;
+			// TX - RX
+			// active-high level-triggered
+			interrupts = <8 4>, <9 4>;
+			// TX and RX
+			dma-channels = <2>;
+
+			status="disabled";
+		};
+
+		axi-ethernet-subsystem@40c00000 {
+			reg = <0x40c00000 0x40000>;
+			compatible = "xlnx,axi-ethernet-subsystem-7.2";
```

Comment:
All non-essential parts for the device to operate should be disabled, apply throughout